src/core/utils/api-client/notifications.ts

```diff
@@ -22,25 +23,37 @@ export class NotificationsApiClient {
      * TODO: When fetching multiple pages of notifications, maybe we should find a way to cache those notifications
      *       in order to minimise the calls to our api
      */
-    public async fetchNotifications(walletPublicKey: string, page?: number) {
+    public async fetchNotifications(
+        walletPublicKeys: string[],
+        blockchainNetworks: { blockchain: Blockchain; chainId: string }[],
+        page?: number
+    ) {
         try {
-            const walletPrivateKey = await getWalletCredentialsKey(walletPublicKey);
-            if (walletPrivateKey) {
-                const data: any = {
+            const data = {
+                walletPublicKeys: [],
+                blockchainNetworks,
+                page: page || 1
+            };
+
+            for (const walletPublicKey of walletPublicKeys) {
+                const walletPrivateKey = await getWalletCredentialsKey(walletPublicKey);
+
+                const walletData: any = {
                     walletPublicKey,
                     timestamp: await getCurrentTimestampNTP(),
-                    domain: getWalletApiDomain(),
-                    page: page || 1
+                    domain: getWalletApiDomain()
                 };
 
                 const signature = getSignature(data, walletPrivateKey, walletPublicKey);
```

Comment:
you should use walletData for sign, right ? it's weird to sign all data object...